tests/queries/0_stateless/02843_dont_validate_types_in_cast_on_attach.sql

```diff
@@ -0,0 +1,8 @@
+set allow_experimental_object_type=1;
+drop table if exists test_json;
+create table test_json (x UInt64, json JSON default CAST('{}', 'JSON')) engine=MergeTree order by x;
+detach table test_json;
+set allow_experimental_object_type=1;
```

Comment:
```suggestion
set allow_experimental_object_type=0;
```
This way I think the test makes more sense.